packages/app/src/cli/models/extensions/extension-instance.ts

```diff
@@ -198,6 +209,24 @@ export class ExtensionInstance<TConfiguration extends BaseConfigType = BaseConfi
     return config.build.command
   }
 
+  get watchPaths() {
+    const config = this.configuration as unknown as FunctionConfigType
+    const configuredPaths = config.build.watch ? [config.build.watch].flat() : []
+
+    if (!this.isJavaScript && configuredPaths.length === 0) {
+      return null
+    }
+
+    const watchPaths: string[] = configuredPaths ?? []
+    if (this.isJavaScript && configuredPaths.length === 0) {
+      watchPaths.push(joinPath('src', '**', '*.js'))
+      watchPaths.push(joinPath('src', '**', '*.ts'))
+    }
+    watchPaths.push(joinPath('**', 'input*.graphql'))
```

Comment:
I had meant to check in with you on the input query naming -- I used `**/input*.graphql` with the assumption that there could be multiple input queries, prefixed with `input.`. We can update this if you think this is a bad assumption.

TOML changes are already picked up and deployed by another watcher that already existed for all draftable extensions. 👍

Reply:
I thought we had talked about this as a reasonable default for input query file naming -- `input.targetsuffix.graphql`.   

The fact that we have the config and can auto-watch based on that is interesting. (2) Will deploy the updated extension config based on the toml, but won't restart the file watcher -- for now we considered it reasonable that updating the `watch` setting required restarting `dev`, but I don't think that will be as obvious for the target config.

Globbing on all graphql files is problematic for Rust because it updates the `.output.graphql` each time it builds, which sets off an infinite loop. But maybe we could fix that.